src/ol/render/canvas/style.js

```diff
@@ -179,7 +179,11 @@ export function buildRuleSet(rules, context) {
       }
       someMatched = true;
       for (const styleEvaluator of compiledRules[i].styles) {
-        styles.push(styleEvaluator(context));
+        const style = styleEvaluator(context);
+        if (!style) {
+          continue;
+        }
+        styles.push(style);
```

Comment:
This change handles the `null` styles that can be generated if all of the symbolizers are `null` below.